Task/28 fix melee implementation

Cxyu629 opened the pull request:
- Fixed melee implementation.
- Implemented custom team system, removing dependency from godot groups.
- Fixed bugs #21 and #23 
- Enemy `visuals.gd` is now decoupled from `target_provider.gd`
- Fixed player gun visuals

It is now possible to define multiple `MeleeComp`s for different melee attacks, not that I've tested it out. The same applies for `RangedComp`, allowing both friendly fire and regular fires using separate ranged components.

Cxyu629 committed:
fix: remove visual inconsistencies when gun is facing left v.s. right

Cxyu629 committed:
fix: correct melee targeting

- Add new targeting system based on Team enums in `enums.gd`, `target_filter.gd`
- Modify existing systems to adhere to new targeting system
- Move team value from `TargetPriorityProp` to `HitboxComp`
- Fix
  - bug: error message in bullet.gd:27 @ _on_body_entered() #21
  - bug: enemy melee does not trigger if player does not experience knockback #23

Cxyu629 committed:
chore: bump project version

- Additionally cleanup code

deltaMinor reviewed (commented):
The bugs #23 and #21 have been resolved, however the issue of the player melee not applying effects to enemies remain
side-note: the hurtboxes should all be tinted red to differentiate them from hitboxes for debugging purposes

Cxyu629 reviewed (commented):
> The bugs #23 and #21 have been resolved, however the issue of the player melee not applying effects to enemies remain
> side-note: the hurtboxes should all be tinted red to differentiate them from hitboxes for debugging purposes 

Noted on the tint, I'll add that all hurtboxes have ordering of 100 for visibility.

Oops on the melee effects, I assumed it was for turrets, as seen in `MeleeComp`'s `target_filter` property, will change.

On the tint, does it include bullets and explosions?

deltaMinor commented:
yes for explosions, not necessary for bullets I think

we can have separate colours for bullets if needed, but let's just keep it to "anything that applies effects" to be a hurtbox and thus red and anything else with areas and collisions to be a hitbox and be tinted the default blue for now

Cxyu629 committed:
fix: correct Player's melee targets

- Additionally change hurtboxes' colors and ordering for debugging purposes.
- Additionally fix melee only triggering on a single entity.

Cxyu629 commented:
Hurtboxes modified, also corrected melee's cooldown behavior, see 4ea57d8.

deltaMinor reviewed (approved):
looks like should be good
